lib/manifests/manifests.js

```diff
@@ -0,0 +1,230 @@
+'use strict';
+
+const dateFormat = 'YYYY-MM-DDTHH:mm:ss.SSS[Z]';
+const Promise = require('bluebird');
+const smoment = require('../smoment');
+const Hbase = require('../hbase/hbase-client');
+const Logger = require('../logger');
+const log = new Logger({scope : 'manifests'});
+
+/**
+ * Manifests
+ */
+
+const Manifests = function(config) {
+
+  config.logLevel = 2;
+
+  const hbase = new Hbase(config);
+  const MAX_SEQUENCE = 4294967295
+
+  // Store a map of ephemeral to master public keys for quick lookup
+  // to check each incoming validation for a known master public key.
+  var manifests = {};
+
+  function clearCache() {
+    master_keys = {}
+  }
+
+  function deleteKey(ephemeral_public_key) {
+    delete master_keys[ephemeral_public_key]
+  }
+
+  function setKey(ephemeral_public_key, master_public_key) {
+    master_keys[ephemeral_public_key] = master_public_key
+  }
+
+  function verifySignature(manifest) {
+    const sfSequence = '$'
+    const sfPublicKey = 'q'
+    const sfSigningPubKey = 's'
+
+    // Form manifest
+    var sequence_buf = new Buffer(4)
+    sequence_buf.writeUInt32BE(this.sequence)
+    const sequence_bytes = sequence_buf.toJSON().data
+
+    var master_public_bytes = addressCodec.decodeNodePublic(this.master_public_key)
+    const ephemeral_public_bytes = addressCodec.decodeNodePublic(this.ephemeral_public_key)
+    const signature_bytes = new Buffer(this.signature, 'hex').toJSON().data
+
+    var manifest_data = new Buffer('MAN\0').toJSON().data
+    manifest_data = manifest_data.concat(new Buffer(sfSequence).toJSON().data,
+                               sequence_bytes,
+                               new Buffer(sfPublicKey).toJSON().data,
+                               [master_public_bytes.length],
+                               master_public_bytes,
+                               new Buffer(sfSigningPubKey).toJSON().data,
+                               [ephemeral_public_bytes.length],
+                               ephemeral_public_bytes)
+
+    master_public_bytes.shift()
+    if (!Ed25519.verify(manifest_data, signature_bytes, master_public_bytes)) {
+      return false;
+    }
+    return true;
+  };
+
+  /**
+   * saveManifest
+   * save data to hbase
+   */
+
+  function saveManifest(manifest) {
+    const key = [
+      manifest.master_public_key,
```

Comment:
with this key structure, you'd basically have to get every row from the table in order to populate the cache, as there is no way to only get the last sequence for each master key.  But maybe thats fine, its not expected to be a very large list right?